gbpservice/neutron/nsf/core/main.py

```diff
@@ -0,0 +1,705 @@
+                   % (event.identify(), event.worker_attached)))
+        qu = worker[1]
+        qu.put(event)
+
+    def poll_event(self, event, max_times=sys.maxint):
+        LOG.info(_("Adding to pollq - event %s for maxtimes: %d"
+                   % (event.identify(), max_times)))
+        event.max_times = max_times
+        self.pollhandler.add(event)
+
+    def poll_event_done(self, event):
+        LOG.info(_("Poll event %s done.. Adding to pollq"
+                   % (event.identify())))
+        event.id = 'POLL_EVENT_DONE'
+        self.pollhandler.add(event)
+
+    def report_state(self):
+        self.rpc_agents.report_state()
+
+    def timeout(self):
+        self.pollhandler.poll()
+
+    def register_events(self, evs):
+        for ev in evs:
+            LOG.info(_("Registering event %s & handler %s"
+                       % (ev.identify(), identify(ev.handler))))
+            self.ehs.register(ev)
+
+    def register_rpc_agents(self, agents):
+        self.rpc_agents.add(agents)
+
+    def event(self, **kwargs):
+        return Event(**kwargs)
+
+    def init_complete(self):
+        for module in self.modules:
+            LOG.info(_("Invoking init_complete() of module %s"
+                       % (identify(module))))
+            try:
+                module.init_complete(self, self._conf)
+            except AttributeError:
+                LOG.info(_("Module %s does not implement"
+                           "init_complete() method - skipping"
+                           % (identify(module))))
+
+    def unit_test(self):
+        for module in self.modules:
+            module.unit_test(self._conf, self)
+
+
+def modules_import():
+    modules = []
+    # os.path.realpath(__file__)
+    base_module = __import__(cfg.CONF.modules_dir,
+                             globals(), locals(), ['modules'], -1)
+    # modules_dir = os.getcwd() + "/../modules"
+    modules_dir = base_module.__path__[0]
+    syspath = sys.path
+    sys.path = [modules_dir] + syspath
+
+    try:
+        files = os.listdir(modules_dir)
+    except OSError:
+        LOG.error(_("Failed to read files.."))
+        files = []
+
+    for fname in files:
+        if fname.endswith(".py") and fname != '__init__.py':
+            module = __import__(cfg.CONF.modules_dir,
+                                globals(), locals(), [fname[:-3]], -1)
+            modules += [eval('module.%s' % (fname[:-3]))]
+            # modules += [__import__(fname[:-3])]
+    sys.path = syspath
+    return modules
+
+
+def main():
+    cfg.CONF.register_opts(core_cfg.OPTS)
+    config.register_interface_driver_opts_helper(cfg.CONF)
+    config.register_agent_state_opts_helper(cfg.CONF)
+    config.register_root_helper(cfg.CONF)
+
+    common_config.init(sys.argv[1:])
+    config.setup_logging()
+    modules = modules_import()
+
+    sc = ServiceController(cfg.CONF, modules)
+    sc.start()
+    sc.init_complete()
+    # sc.unit_test()
+    sc.wait()
```

Comment:
Need comments through out this file.
Is this file too big? Is there a coding guideline for file size? If this is true can it be broken up?